win32ss/user/ntuser/message.c

```diff
@@ -2721,6 +2767,12 @@ NtUserMessageCall( HWND hWnd,
                         }
                         ExFreePoolWithTag(List, USERTAG_WINDOWLIST);
                      }
+                     if (lParam && !wParam && wcscmp((WCHAR*)lParam, L"Environment") == 0)
```

Comment:
Should this comparison be done case-sensitively, or not? (i.e. if the string `"enVIroNmEnT"` is passed instead, should the broadcast still be done successfully or not?)

Reply:
The string is not case sensitive. Not that it matters, all strings are supposed to be supported. Currently it's missing "SystemDockMode" and "ConvertibleSlateMode" (and all possible registry key names).

Reply:
Where does this lParam parameter come from?

Reply:
> Where does this lParam parameter come from?

When the message is WM_SETTINGCHANGE, lparam can be NULL or a string that says which setting you changed. There are maybe about 7-8 strings that are "standard" but it can also be the name of a registry key.

Reply:
> checking for "Environment" seems like a hack to me.

I agree.

Reply:
In accordance with the title of this PR, it was intended to only handle ```Environment`` strings. It can be expanded later in another PR. Anything else is "out of scope" for this PR. Thanks.

Reply:
@whindsaks, I see that you have given a "thumbs down" comment here. I just do not have the background in this area of the code to do much more with it for now. Are you willing to pick this up? If so, you are free to make any changes here that you want. Or if you want to create a completely new PR, I can close this one. I do appreciate your willingness to help here and want to encourage you in any way that I can. Thanks.

Reply:
I don't know much about the kernel part of the window manager either, I just feel that we should try to make it work with all strings from the start, not a hack like this for just one string.